loader/src/injector/hook.cpp

```diff
@@ -49,9 +48,14 @@ enum {
 
 #define DCL_PRE_POST(name) \
 void name##_pre();         \
-void name##_post();
+void name##_post()
+
+#define IS_FLAG_SET(flags_variable, flags_to_check) \
+    ((flags_variable & flags_to_check) == flags_to_check)
 
 #define MAX_FD_SIZE 1024
+#define DIR_FD "/proc/self/fd"
+#define MEM_FD "/memfd:/boot-image-methods.art"
```

Comment:
I was trying to more closely match the original code from AOSP, so it's better without a define.

Reply:
It was an effort to reduce boilerplate. The `fd` path in `DIR_FD` is referenced 4 times, increasing the odds of a mistake.
Original code from AOSP is also not the best source for proper coding practices.